crates/wmemcheck/src/lib.rs

```diff
@@ -140,6 +158,9 @@ impl Wmemcheck {
 
     /// Updates memory checker memory state metadata when free is called.
     pub fn free(&mut self, addr: usize) -> Result<(), AccessError> {
+        if addr == 0 {
```

Comment:
Good find! Can we add a comment here noting the implemented semantics (`free` of a null pointer is a no-op)?